### ISSUE TITLE
Documentation for unarchive module does not match remote URL behaviour

### ISSUE THREAD

**Issue by acdha** [ansible/ansible-modules-core#4433]
##### ISSUE TYPE
- Documentation Report
##### COMPONENT NAME

unarchive
##### ANSIBLE VERSION

```
ansible 2.1.1.0
  config file = 
  configured module search path = Default w/o overrides
```

(via Homebrew)
##### CONFIGURATION

n/a
##### OS / ENVIRONMENT

OS X managing RHEL 6
##### SUMMARY

https://docs.ansible.com/ansible/unarchive_module.html has an example like this:

```
# Unarchive a file that needs to be downloaded (added in 2.0)
- unarchive: src=https://example.com/example.zip dest=/usr/local/bin remote_src=yes
```

That doesn't match the behaviour observed with 2.1.1.0. It's not immediately obvious whether this is the same root cause as #3228 but since that was reported against 2.0 and closed as fixed in 2.1, and the error message is different, I'm opening this issue.
##### STEPS TO REPRODUCE

```
    - name: install Prometheus distributable
      unarchive:
        src: https://github.com/prometheus/prometheus/releases/download/v1.0.1/prometheus-1.0.1.linux-amd64.tar.gz
        dest: "{{ prometheus_work_directory }}"
        remote_src: yes
```
##### EXPECTED RESULTS

The binary would be downloaded and unarchived
##### ACTUAL RESULTS

Fails with an error showing the remote URL used as a local filename:

```
<server.example.org> PUT /Users/cadams/Projects/lc-prometheus-server/https:/github.com/prometheus/prometheus/releases/download/v1.0.1/prometheus-1.0.1.linux-amd64.tar.gz TO /tmp/ansible-tmp-1471296081.16-241397141036424/source
fatal: [server.example.org]: FAILED! => {"failed": true, "msg": "file or module does not exist: /Users/cadams/Projects/lc-prometheus-server/https:/github.com/prometheus/prometheus/releases/download/v1.0.1/prometheus-1.0.1.linux-amd64.tar.gz"}
```

Using `copy: no` instead of `remote_src: yes` works correctly but is described as deprecated.

**Comment by ansibot** [comment]
@dagwieers, @pileofrogs, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dagwieers** [comment]
The **remote_src** option is to be introduced for Ansible v2.2, unfortunately the documentation was updated but is not versioned for specific versions. So the examples are already using the new options, whereas those are not yet available in v2.1 or older. This is a known issue with the documentation in general. (It reflects the devel branch, not any of the stable branches...)

Also see: ansible/ansible#13858

So this is not something to fix in the **unarchive** module, but a more general issue with the documentation. wontfix duplicate_of ansible/ansible#13858
